### PR TITLE
adding ext-switch entity definition and metrics

### PR DESCRIPTION
**Pull request by thezackm**
### Relevant information

adding `SWITCH` entity for the `EXT` domain. Original definition is based on network device metrics sent via Kentik technical partnership, but this definition is being kept ambiguous to allow for other technical partnerships to use the entity in the future.

`SWITCH` relates to any L2 device in the networking space that lacks the capability to route packets.

### Checklist

* [X] I've read the guidelines and understand the acceptance criteria.
* [X] The value of the attribute marked as `identifier` will be unique and valid. 
* [X] I've confirmed that my entity type wasn't already defined. If it is I'm providing an
 explanation above.

### CODE REVIEW

File: definitions/ext-switch/golden_metrics.yml
Comment (srvaroa) [inline definitions/ext-switch/golden_metrics.yml:6]: The `where` clauses should not be necessary I think as the GMs will match against telemetry that is already stamped with a GUID.  Is that right @naxhh ?

**Review by srvaroa — COMMENTED** [review]
LGTM, just a minor question pending for @naxhh 's feedback.

File: definitions/ext-switch/golden_metrics.yml
Comment (IreneMLC) [reply definitions/ext-switch/golden_metrics.yml:6]: I think that for that to happen you'd have to use [queries](https://github.com/newrelic-experimental/entity-synthesis-definitions/blob/main/docs/summary_metrics.md#roll-up-entities), it might make sense here:

> In the cases that the entity type can be ingested from different sources and you need to provide a different query implementation you can use queries instead of query

Comment (thezackm) [reply definitions/ext-switch/golden_metrics.yml:6]: for context; this is using the same pattern as multiple [AWS Entities](https://github.com/newrelic-experimental/entity-synthesis-definitions/blob/main/definitions/infra-awsalb/golden_metrics.yml) in that `provider` can have multiple values so the `WHERE` seems helpful.

If it's not required because there's an implicit `WHERE entityGuid = '1234abcd'`; then I would agree it's unnecessary.

Comment (srvaroa) [reply definitions/ext-switch/golden_metrics.yml:6]: Yeah, when the queries get executed they will be applied to a specific entity, so the `where` would be redundant. But it doesn't break anything anyway.

Comment (thezackm) [reply definitions/ext-switch/golden_metrics.yml:6]: ok; would you like me to remove them @srvaroa? I can get that done real quick so this can get merged.

Comment (thezackm) [reply definitions/ext-switch/golden_metrics.yml:6]: also; would the same apply to the Summary Metrics?

Comment (srvaroa) [reply definitions/ext-switch/golden_metrics.yml:6]: Yes! I'll ping @naxhh  to get this approved soon as possible on our end.

Comment (srvaroa) [reply definitions/ext-switch/golden_metrics.yml:6]: @thezackm let's leave it for now, we're not 100% sure if that where is important for golden signals.  I can clean it up later if needed.

Comment (naxhh) [reply definitions/ext-switch/golden_metrics.yml:6]: As talking with @srvaroa I have a doubt on a specific use case. So let's keep the provider for now and I'll double check that.

**Review by naxhh — APPROVED** [review]
LGTM.

We need to double check some behaviors in our side before merging

**Review by srvaroa — APPROVED** [review]

**Comment by srvaroa** [comment]
@thezackm have you signed the CLA? (I have seen the bot failed to report status on several PRs so it may be that it just broke).

**Comment by thezackm** [comment]
@srvaroa  yes I have. I agree it might be broken, this happened to me on another repo recently as well.

**Pull request merged**
